compiler/src/iree/compiler/Codegen/Common/TransformDialectStrategies.cpp

```diff
@@ -303,24 +312,40 @@ static Value buildReductionStrategyThreadDistributionPart(
   return variantH;
 }
 
+/// Returns a pair of handles to the main reduction operation and the fusion
+/// root, which may or may not be equal. When not equal, there are users of
+/// reduction results into which the reduction may be fused.
+static std::tuple<Value, Value> reductionBlockDistributionHandles(
```

Comment:
I see that this is doing some internal notional `bind` like I typed above and ends up resolving later with equality comparison on the 2 handles.

I think bind + value == null generalizes a bit better than 1-off reliance on custom contracts involving equality comparison.

Reply:
Done.